workers/unity/Packages/com.improbable.gdk.tools/RedirectedProcess.cs

```diff
@@ -95,6 +104,89 @@ void OnReceived(object sender, DataReceivedEventArgs args)
             }
         }
 
+        /// <summary>
+        ///     Runs the redirected process and returns a task which can be waited on.
+        /// </summary>
+        /// <param name="workingDirectory">The directory to run the command from.</param>
+        /// <param name="command">The filename to run.</param>
+        /// <param name="arguments">Parameters that will be passed to the command.</param>
+        /// <param name="redirectStdout">Redirect standard output to Debug.Log</param>
+        /// <param name="redirectStderr">Redirect standard error to Debug.LogError</param>
+        /// <returns>A task which would return the exit code and output.</returns>
+        public static async Task<RedirectedProcessResult> RunInAsync(string workingDirectory, string command, string[] arguments, bool redirectStdout = false, bool redirectStderr = false)
+        {
+            var info = new ProcessStartInfo(command, string.Join(" ", arguments))
+            {
+                CreateNoWindow = true,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                UseShellExecute = false,
+                WorkingDirectory = workingDirectory
+            };
+
+            return await Task.Run(() =>
+            {
+                using (var process = Process.Start(info))
+                {
+                    if (process == null)
+                    {
+                        throw new Exception(
+                            $"Failed to run {info.FileName} {info.Arguments}\nIs the .NET Core SDK installed?");
+                    }
+
+                    process.EnableRaisingEvents = true;
+
+                    var processStandardOutput = new List<string>();
```

Comment:
A `StringBuilder` will be much more efficient.

Reply:
Yeah agreed (that's what used to be there). However, the design decision I made here is that I wanted to make sure that separate lines of output were in a list rather than CRLF characters in the string itself. As it's not exactly expensive to convert a collection of strings to a single string with `string.Join` (or whatever it is) for general output sizes of 10's of lines of output, I figured it was worth it, to avoid having to split a single string into multiple strings later.

Reply:
Why do you need individual lines of text?

Reply:
In the deployment manager thing I wrote in a different PR, I need to invoke a binary using the Platform SDK to obtain a list of deployments. This binary (an invocation of `dotnet run`) outputs a list of deployment_name deployment_id pairs, and it's far easier to process this without having to split across newlines.